tests/testversion.pas

```diff
@@ -0,0 +1,54 @@
+program testversion;
+
+{
+
+  Test version macros/functions of SDL2 system.
+
+  This file is part of
+
+    SDL2-for-Pascal
+    Copyright (C) 2020-2022 PGD Community
+    Visit: https://github.com/PascalGameDevelopment/SDL2-for-Pascal
+
+}
+
+{$I testsettings.inc}
+
+uses
+  Classes, SysUtils, SDL2;
+
+type
+  ESDL2Error = class(Exception);
+
+var
+  CompiledVersion: TSDL_Version = (major: 0; minor: 0; patch: 0);
+  LinkedVersion: TSDL_Version = (major: 0; minor: 0; patch: 0);
+begin
+  write('Start SDL2 version test... ');
+
+  try
+    if (SDL_VERSIONNUM(1,2,3) <> 1203) then
+      raise ESDL2Error.Create('SDL_VERSIONNUM failed: 1203 expected, found: ' + IntToStr(SDL_VERSIONNUM(1,2,3)));
+
+    SDL_VERSION(CompiledVersion);
+    if (SDL_COMPILEDVERSION <> SDL_VERSIONNUM(CompiledVersion.major, CompiledVersion.minor, CompiledVersion.patch)) then
+      raise ESDL2Error.Create('SDL_VERSION or SDL_COMPILEDVERSION failed: Version results do not match!');
+
+    if (SDL_VERSION_ATLEAST(2,0,0) <> True) then
```

Comment:
Personally I really dislike comparisons with boolean literals (unless we're talking about dynamically-typed languages). I'd simplify this to just:
```pascal
if not SDL_VERSION_ATLEAST(2,0,0) then
```